File: backend/src/Organi.Server/Organi.Server.Domain/Entities/Vendor.cs
using Organi.Server.Domain.Common;
using Organi.Server.Domain.Enums;

namespace Organi.Server.Domain.Entities;

public sealed class Vendor : AuditableEntity
{
    public string StoreName { get; set; } = default!;
    public string Slug { get; set; } = default!;
    public string? Description { get; set; }
    public string? LogoUrl { get; set; }
    public string? BannerUrl { get; set; }
    public string? PhoneNumber { get; set; }
    public string? Address { get; set; }
    public string? City { get; set; }
    public decimal Rating { get; set; }
    public int TotalSales { get; set; }
    public int FollowerCount { get; set; }
    public VendorStatus Status { get; set; } = VendorStatus.Pending;
    public Guid UserId { get; set; }

    public User User { get; set; } = default!;
    public ICollection<Product> Products { get; set; } = [];
}

File: frontend/web/src/types/api/vendor.ts
// Mirrors Organi.Server.Domain.Enums.VendorStatus (serialized as string)
export type VendorStatus = 'Pending' | 'Approved' | 'Suspended' | 'Rejected'

// Mirrors Organi.Server.Application.Features.Vendors.DTOs.VendorResponse
export interface VendorResponse {
  id: string
  storeName: string
  slug: string
  description: string | null
  logoUrl: string | null
  bannerUrl: string | null
  phoneNumber: string | null
  address: string | null
  city: string | null
  rating: number
  totalSales: number
  followerCount: number
  status: VendorStatus
  createdAt: string
  updatedAt: string | null
}
